src/test/java/wellnus/atomichabit/AtomicHabitTest.java

```diff
@@ -1,33 +1,76 @@
 package wellnus.atomichabit;
 
 import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import wellnus.atomichabit.command.AddCommand;
-import wellnus.atomichabit.command.Command;
 import wellnus.atomichabit.command.UpdateCommand;
+import wellnus.command.Command;
+import wellnus.command.CommandParser;
 import wellnus.exception.AtomicHabitException;
 import wellnus.atomichabit.feature.AtomicHabitList;
 import wellnus.atomichabit.feature.AtomicHabitManager;
+import wellnus.exception.WellNusException;
+import wellnus.ui.TextUi;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.HashMap;
 
 public class AtomicHabitTest {
-    private AtomicHabitList habitList;
+    private static final String UPDATE_HABIT_COMMAND = "hb update";
+    private final AtomicHabitList habitList;
+    private final ByteArrayOutputStream outputStreamCaptor;
+    private final CommandParser parser;
+
+    public AtomicHabitTest() {
+        this.habitList = new AtomicHabitList();
+        this.outputStreamCaptor = new ByteArrayOutputStream();
+        this.parser = new CommandParser();
+    }
+
+    private String getMessageFrom(String uiOutput) {
+        TextUi textUi = new TextUi();
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outputStream));
+        textUi.printSeparator();
+        String separator = outputStream.toString().trim();
+        StringBuilder resultBuilder = new StringBuilder();
+        String[] outputLines = uiOutput.split(System.lineSeparator());
+        for (String outputLine : outputLines) {
+            String trimmedOutputLine = outputLine.trim();
+            if (!trimmedOutputLine.equals(separator)) {
+                resultBuilder.append(trimmedOutputLine).append(System.lineSeparator());
+            }
+        }
+        return resultBuilder.toString().trim();
+    }
+
+    @BeforeEach
+    public void setUp() {
+        System.setOut(new PrintStream(outputStreamCaptor));
+    }
 
     /**
      * Test AddCommand with a standard payload to check output printed
      */
     @Test
-    public void addHabit_checkOutput_success() throws AtomicHabitException {
-        habitList = new AtomicHabitList();
+    public void addHabit_checkOutput_success() throws WellNusException {
         String payload = "junit test";
         String expectedOutput = "Yay! You have added a new habit:"
                 + System.lineSeparator()
                 + "'"
                 + payload
                 + "'"
                 + " was successfully added";
-        Command command = new AddCommand(payload);
-        Assertions.assertEquals(expectedOutput, command.execute(habitList).getCommandResult());
+        String testCommand = "hb add --name " + payload;
```

Comment:
May I check are we using feature name in command? iirc we said we gonna remove all the feature names in commands